plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -547,6 +552,10 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                      fc_dlog( _trx_trace_log, "[TRX_TRACE] Speculative execution COULD NOT FIT tx: ${txid} RETRYING",
                               ("txid", trx->id()));
                   }
+                  if( block_is_exhausted() ) {
+                     schedule_maybe_produce_block( true );
```

Comment:
This should be called in `on_incoming_transaction_async` instead if `process_incoming_transaction_async` returns false and `_pending_block_mode == pending_block_mode::producing`.

Reply:
Nice. Done.